### ISSUE TITLE
sr.py status broken by inconsistent directories.

### ISSUE THREAD

**Issue by petersilva** [MetPX/sarracenia#295]
in some cases, when .config/sarra and .cache/sarra have inconsistent states (say a configuration without a state, or a state without a configuration) sr will crash.

**Comment by petersilva** [comment]
sample failure:

```
blacklab% sr status
reading procs: ..... Done reading 526 procs!
gathering global state: procs, configs, state files, logs, analysis - Done. 
status 
Component  State      Good?  Qty Configurations-i(r/e)-r(Retry)
---------  -----      -----  --- ------------------------------
audit      running    OK       1
cpost      mixed      mult     2
      1 stopped: local_post 
      1 running: veille_f34-i1/1 
cpump      mixed      mult     4
      2 stopped: pelle_dd2_f05, pelle_dd1_f04 
      2 running: xvan_f15-i1/1, xvan_f14-i1/1 
poll       running    OK       2 pulse-i1/1, f62-i1/1
Traceback (most recent call last):
  File "/usr/bin/sr", line 11, in <module>
    load_entry_point('metpx-sarracenia==2.20.1b1', 'console_scripts', 'sr')()
  File "/home/peter/src/sarracenia/sarra/sr.py", line 736, in main
    sys.exit(gs.status())
  File "/home/peter/src/sarracenia/sarra/sr.py", line 662, in status
    if self.states[c][cfg]['retry_queue'] > 0 :
KeyError: 'post'
blacklab% 

```

**Comment by petersilva** [comment]
reported by @habilinour, included in 2.19.12b7
